### ISSUE TITLE
AutoLayout with OpenOrd and forceAtlas2 doesn't work

### ISSUE THREAD

**Issue by ZhitongLei** [gephi/gephi#1571]
I wanna draw a graph with OpenOrdLayout and forceAtlas2, after autoLayout.execute(), it seems that the process is blocked. Is there anything wrong in my program? Here's my code:

`GraphModel graphModel = Lookup.getDefault().lookup(GraphController.class).getModel();
AutoLayout autoLayout = new AutoLayout(1, TimeUnit.MINUTES);
autoLayout.setGraphModel(graphModel);
OpenOrdLayout firstLayout = new OpenOrdLayout(new OpenOrdLayoutBuilder());
ForceAtlas2 secondLayout = new ForceAtlas2(new ForceAtlas2Builder());
AutoLayout.DynamicProperty adjustBySize = AutoLayout.createDynamicProperty("ForceAtlas2.adjustSizes.name", Boolean.TRUE, 0.9f); //True after 90% of layout time
autoLayout.addLayout(firstLayout, 0.5f);
autoLayout.addLayout(secondLayout, 0.5f, new AutoLayout.DynamicProperty[]{adjustBySize});
autoLayout.execute();`

**Comment by eduramiba** [comment]
Is it blocked after 1 minute?

**Comment by ZhitongLei** [comment]
@eduramiba yes ,it blocked after 1 minute, doesn't  consume CPU anymore. It print this(the ouput of openord) then block

##
    Real iterations 150
    Entering liquid stage...
    Liquid stage completed in 3 seconds, total energy = 31920807936.000000
    Entering expansion stage...
    Expansion stage completed in 2 seconds, total energy = 21202154.000000
    Entering cool-down stage...
    Cool-down stage completed in 3 seconds, total energy = 979824.875000
    Entering crunch stage...
    Crunch stage completed in 1 seconds, total energy = 1019501.875000
    Entering simmer stage...
    Simmer stage completed in 0 seconds, total energy = 10.938383
    Layout completed in 9 seconds with 752 iterations

**Comment by eduramiba** [comment]
And what version of the toolkit are you using?

**Comment by ZhitongLei** [comment]
@eduramiba toolkit version is 0.8.2, java version is 1.8,
should I use the latest version of toolkit?

**Comment by eduramiba** [comment]
No, 0.8.2 should be fine as I am not sure 0.9.1 is ready for autolayout.

**Comment by ZhitongLei** [comment]
@eduramiba If I add System.exit(0) at the end of the program, it will exit normally,  is this means that the main thread is not blocked, only the thread of autoLayout is blocked?

**Comment by eduramiba** [comment]
Oh, that means there is some thread not finished. This was kind of a known issue with Gephi 0.8.2, where some threads are left unfinished, preventing the JVM from exiting.

This should not happen with 0.9.1 toolkit, you can try that. Otherwise I guess you can use that System.exit workaround.

**Comment by ZhitongLei** [comment]
When using 0.9.1 toolkit, the program has the same behavior as  0.8.2.

And I found that when duration is large than 1 minute, System.exit(0) is useless, the program will not exit normally(both 0.9.1 and 0.8.2).

So I don't know what to do now.  =_=
Thanks anyway.

**Comment by eduramiba** [comment]
We will need to debug this problem in 0.9.1 then to find out what thread is left. A thread dump from jvisualvm will probably help.

**Comment by axololtl1997** [comment]
Hello,
This error seems to have continued onto v0.9.2 of the gephi toolkit? Any ETA on when it will be fixed?
Thanks, A.K.
